api/auth/azure/azure.go

```diff
@@ -0,0 +1,204 @@
+package azure
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/hashicorp/vault/api"
+)
+
+type AzureAuth struct {
+	roleName  string
+	mountPath string
+	resource  string
+}
+
+var _ api.AuthMethod = (*AzureAuth)(nil)
+
+type LoginOption func(a *AzureAuth) error
+
+type responseJSON struct {
+	AccessToken  string `json:"access_token"`
+	RefreshToken string `json:"refresh_token"`
+	ExpiresIn    string `json:"expires_in"`
+	ExpiresOn    string `json:"expires_on"`
+	NotBefore    string `json:"not_before"`
+	Resource     string `json:"resource"`
+	TokenType    string `json:"token_type"`
+}
+
+type metadataJSON struct {
+	Compute computeJSON `json:"compute"`
+}
+
+type computeJSON struct {
+	VMName            string `json:"name"`
+	VMScaleSetName    string `json:"vmScaleSetName"`
+	SubscriptionID    string `json:"subscriptionId"`
+	ResourceGroupName string `json:"resourceGroupName"`
+}
+
+const (
```

Comment:
This seems to be duplicating the `go-autorest` definitions provided by azure, see [here](https://github.com/Azure/go-autorest/blob/master/autorest/azure/environments.go). IMO the UX is much better if the user can just provide a struct property instead of having to fiddle around with strings to construct the resource URI etc.

Reply:
This is my first time working with Azure, so thank you for making me aware of this repo!